src/identity-providers/routes/IdentityProvider.ts

```diff
@@ -1,20 +1,32 @@
 import type { LocationDescriptorObject } from "history";
 import { generatePath } from "react-router-dom";
 import type { RouteDef } from "../../route-config";
-import {
-  AddIdentityProvider,
-  IdentityProviderCrumb,
-} from "../add/AddIdentityProvider";
+import { AddIdentityProvider } from "../add/AddIdentityProvider";
+import { DetailSettings } from "../add/DetailSettings";
 
-export type IdentityProviderParams = {
+type IdentityProviderTabs = "settings" | "mappers";
+
+export type IdentityProviderCreateParams = {
   realm: string;
-  id: string;
+  providerId?: string;
+};
+
+export type IdentityProviderParams = IdentityProviderCreateParams & {
+  alias: string;
+  tab: IdentityProviderTabs;
```

Comment:
Each route should have no dependencies on other routes, and neither should their types. The reason for this is to prevent accidentally introducing new params when editing another route.
```suggestion
export type IdentityProviderParams = {
  realm: string;
  providerId: string;
  alias: string;
  tab: IdentityProviderTabs;
```